### ISSUE TITLE
Startup crash on Windows7

### ISSUE THREAD

**Issue by afarlie** [DISTRHO/Cardinal#310]
Installed - [Cardinal-win32-22.06-installer.exe](https://github.com/DISTRHO/Cardinal/releases/download/22.06/Cardinal-win32-22.06-installer.exe) 

Attempted to start the program with a JACK server already running.

From the windows crash dialog : -
Problem signature:
  Problem Event Name:	APPCRASH
  Application Name:	Cardinal.exe
  Application Version:	0.0.0.0
  Application Timestamp:	62bc9236
  Fault Module Name:	Cardinal.exe
  Fault Module Version:	0.0.0.0
  Fault Module Timestamp:	62bc9236
  Exception Code:	c0000005
  Exception Offset:	008fe713
  OS Version:	6.1.7601.2.1.0.256.48
  Locale ID:	2057
  Additional Information 1:	0a9e
  Additional Information 2:	0a9e372d3b4ad19135b953a78882e789
  Additional Information 3:	0a9e
  Additional Information 4:	0a9e372d3b4ad19135b953a78882e789

**Comment by falkTX** [comment]
Does your system have graphics drivers installed? Can it do OpenGL?

**Comment by afarlie** [comment]
Yes I have grpahics drivers installed, but the chipset doesn't support the most recent OpenGL version, as it's a much older chipset.

Extract from DXdiag output:-
```

        Card name: Intel(R) G41 Express Chipset
       Manufacturer: Intel Corporation
          Chip type: Intel(R) 4 Series Express Chipset Family
           DAC type: Internal
     Display Memory: 1453 MB
   Dedicated Memory: 64 MB
      Shared Memory: 1389 MB
       Current Mode: 1440 x 900 (32 bit) (60Hz)
       Monitor Name:  Hanns.G HW191
      Monitor Model: Hanns.G HW191
         Monitor Id: HSD8991
        Native Mode: 1440 x 900(p) (59.887Hz)
        Output Type: HD15
        Driver Name: igdumdx32.dll,igd10umd32.dll
Driver File Version: 8.15.0010.2869 (English)
     Driver Version: 8.15.10.2869
        DDI Version: 10
       Driver Model: WDDM 1.1
  Driver Attributes: Final Retail
   Driver Date/Size: 11/8/2012 15:26:08, 571904 bytes
        WHQL Logo'd: n/a
    WHQL Date Stamp: n/a
          Vendor ID: 0x8086
          Device ID: 0x2E32
          SubSys ID: 0x04001028
        Revision ID: 0x0003
 Driver Strong Name: oem24.inf:IntelGfx.NTx86.6.0:iEGL0:8.15.10.2869:pci\ven_8086&dev_2e32&subsys_04001028
     Rank Of Driver: 00E60001
        Video Accel: ModeMPEG2_A ModeMPEG2_C ModeWMV9_B ModeWMV9_C ModeVC1_B ModeVC1_C 
       D3D9 Overlay: Supported
            DXVA-HD: Supported
       DDraw Status: Enabled
         D3D Status: Enabled
         AGP Status: Enabled

```

**Comment by dromer** [comment]
From what I can tell this is mainly a driver issue, but it does seem to be an old and under-powered GPU.
Are you able to run Rack2Free on this?

**Comment by afarlie** [comment]
Rack2Free doesn't support Windows 7 (or it needs the 64 bit version) as it complained that I had the wrong Windows version.

**Comment by falkTX** [comment]
Does the "makehuman" thing work for you?
I can add the same workarounds, but only if they actually work, otherwise it is a waste of time.

Cardinal for sure shouldn't be crashing, I can fix that one.

**Comment by falkTX** [comment]
it cant. so window just stays black in that case.

**Comment by afarlie** [comment]
The makehuman code works for me. (or the version in which the GL fallbacks were implemented does.)  It would of course be acceptable the exception, and 'fail' gracefully if there's an issue with older OpenGL contexts.

I am of course not saying that this will solve the issue, given that I've had jackd crash on me when some clients have tried to connect to it.

**Comment by falkTX** [comment]
ok good. I added the fallback functions just now, will trigger a build for it later on.
this will be part of the 22.07 release a bit later today.

**Comment by falkTX** [comment]
ok please try with the win32 builds in https://github.com/DISTRHO/Cardinal/releases/tag/22.07
hopefully that works now!

**Comment by afarlie** [comment]
Using the most recent build -
Problem signature:
"  Problem Event Name:	APPCRASH
  Application Name:	Cardinal.exe
  Application Version:	0.0.0.0
  Application Timestamp:	62d1d7c3
  Fault Module Name:	Cardinal.exe
  Fault Module Version:	0.0.0.0
  Fault Module Timestamp:	62d1d7c3
  Exception Code:	c0000005
  Exception Offset:	0038bf56
  OS Version:	6.1.7601.2.1.0.256.48
  Locale ID:	2057
  Additional Information 1:	0a9e
  Additional Information 2:	0a9e372d3b4ad19135b953a78882e789
  Additional Information 3:	0a9e
  Additional Information 4:	0a9e372d3b4ad19135b953a78882e789
"

**Comment by afarlie** [comment]
Also reporting - 
"Problem signature:
  Problem Event Name:	APPCRASH
  Application Name:	Carla.exe
  Application Version:	0.0.0.0
  Application Timestamp:	625951bd
  Fault Module Name:	CardinalSynth.dll
  Fault Module Version:	0.0.0.0
  Fault Module Timestamp:	62d1e33d
  Exception Code:	c0000005
  Exception Offset:	00388f96
  OS Version:	6.1.7601.2.1.0.256.48
  Locale ID:	2057
  Additional Information 1:	0a9e
  Additional Information 2:	0a9e372d3b4ad19135b953a78882e789
  Additional Information 3:	0a9e
  Additional Information 4:	0a9e372d3b4ad19135b953a78882e789"

When I attempted to use the Carla based plugin as an alternative.

Something is screwed up here, and I lack the expertise to determine what.
